nemo/collections/asr/parts/preprocessing/segment.py

```diff
@@ -199,51 +201,111 @@ def from_file(
         :param channel selector: string denoting the downmix mode, an integer denoting the channel to be selected, or an iterable
                                  of integers denoting a subset of channels. Channel selector is using zero-based indexing.
                                  If set to `None`, the original signal will be used.
-        :return: numpy array of samples
+        :return: AudioSegment instance
         """
         samples = None
-        if not isinstance(audio_file, str) or os.path.splitext(audio_file)[-1] in sf_supported_formats:
-            try:
-                with sf.SoundFile(audio_file, 'r') as f:
-                    dtype = 'int32' if int_values else 'float32'
-                    sample_rate = f.samplerate
-                    if offset > 0:
-                        f.seek(int(offset * sample_rate))
-                    if duration > 0:
-                        samples = f.read(int(duration * sample_rate), dtype=dtype)
-                    else:
-                        samples = f.read(dtype=dtype)
-            except RuntimeError as e:
-                logging.error(
-                    f"Loading {audio_file} via SoundFile raised RuntimeError: `{e}`. "
-                    f"NeMo will fallback to loading via pydub."
+        if isinstance(audio_file, list):
+            # If a list of files is provided, load samples from individual single-channel
+            # files and concatenate them along the channel dimension
```

Comment:
What if user names mistakes and send list of two different file here ? Can we detect and raise error?